File: app/src/main/java/com/zayed/admin/yummyrecipes/ProgressbarAnimation.java
package com.zayed.admin.yummyrecipes;

import android.content.Context;
import android.content.Intent;
import android.view.animation.Animation;
import android.view.animation.Transformation;
import android.widget.ProgressBar;
import android.widget.TextView;

public class ProgressbarAnimation extends Animation {

    private Context context;
    private ProgressBar progressBar;
    private TextView textView;
    private float from;
    private float to;

    public ProgressbarAnimation(Context context,ProgressBar progressBar,TextView textView,float from,float to)
    {
        this.context = context;
        this.progressBar = progressBar;
        this.textView = textView;
        this.from = from;
        this.to = to;
    }


    @Override
    protected void applyTransformation(float interpolatedTime, Transformation t) {
        super.applyTransformation(interpolatedTime, t);
        int value = (int) (from + (to - from) * interpolatedTime);
        progressBar.setProgress((int)value);
        textView.setText(value+"%");

        if (value == to)
        {
            context.startActivity(new Intent(context,MainActivity.class));
        }
    }

}

File: app/src/main/java/com/zayed/admin/yummyrecipes/ProgressbarAnimation.kt
package com.zayed.admin.yummyrecipes

import android.content.Context
import android.content.Intent
import android.view.animation.Animation
import android.view.animation.Transformation
import android.widget.ProgressBar
import android.widget.TextView

class ProgressbarAnimation(private val context: Context, private val progressBar: ProgressBar, private val textView: TextView, private val from: Float, private val to: Float) : Animation() {
    override fun applyTransformation(interpolatedTime: Float, t: Transformation) {
        super.applyTransformation(interpolatedTime, t)
        val value = (from + (to - from) * interpolatedTime).toInt()
        progressBar.progress = value
        textView.text = "$value%"
        if (value.toFloat() == to) {
            context.startActivity(Intent(context, MainActivity::class.java))
        }
    }
}